tests/rptest/scale_tests/large_messages_test.py

```diff
@@ -0,0 +1,488 @@
+            running_time_sec = 120
+            # and use scale parameter for max throughput
+            expected_min_throughput = self.scale.expect_single_bandwidth
+
+        # Lower throughput expectation by 10% to account for network errors
+        # and swarm node slow start/end
+        expected_min_throughput *= 0.9
+        expected_throughput_mb = expected_min_throughput / 1024 / 1024
+        self.logger.info(f"Total data: {total_mb:.2f}MB ({total_bytes}b), "
+                         f"Expected throughput >{expected_throughput_mb}MB/s")
+
+        # # Run swarm consumers
+        _group = "large_messages_group"
+        swarm_consumers = self._run_consumers_with_constant_rate(
+            self.n_topics, self.topic_prefixes, _group)
+
+        # Run swarm producers
+        swarm_producers = self._run_producers_with_constant_rate(
+            self.n_topics, self.topic_prefixes,
+            messages_per_second_per_producer)
+
+        # Wait for all messages to be produced
+        self.logger.info("Measuring bandwidth")
+        # Measure bandwidth each 2 seconds
+        # if no new bytes received by RP, check swarm and exit
+        # if at least one finished
+        bandwidth_in = []
+        bandwidth_out = []
+        backoff_interval = 2
+        seconds_spent = 0
+        last_read, last_sent = self._get_rw_metrics()
+        interval_start_sec = time.time()
+        while seconds_spent < running_time_sec:
+            read, sent = self._get_rw_metrics()
+            # Since we are doing some other calculations,
+            # Use timing to determine elapsed interval
+            interval_end_sec = time.time()
+            elapsed_sec = interval_end_sec - interval_start_sec
+            # Calculate ingress BW
+            bytes_per_sec_in = (read - last_read) / elapsed_sec
+            mb_per_sec_in = bytes_per_sec_in / 1024 / 1024
+            # Calculate egress BW
+            bytes_per_sec_out = (sent - last_sent) / 2
+            mb_per_sec_out = bytes_per_sec_out / 1024 / 1024
+            self.logger.debug(
+                f"Bytes read: {read} ({mb_per_sec_in:.2f}MB/sec), "
+                f"Bytes sent: {sent} ({mb_per_sec_out:.2f}MB/sec)")
+            # If no new bytes received, check swarm nodes
+            if last_read == read:
+                if any([not s.is_alive() for s in swarm_producers]):
+                    # At least one is done, exit
+                    break
+
+            # Save measurements
+            bandwidth_in.append(bytes_per_sec_in)
+            bandwidth_out.append(bytes_per_sec_out)
+
+            last_read = read
+            last_sent = sent
+            time.sleep(backoff_interval)
+            seconds_spent += elapsed_sec
+            interval_start_sec = interval_end_sec
+
+        # Run checks if swarm nodes finished
+        self.logger.info("Make sure that swarm node producers are finished")
+        for s in swarm_producers:
+            # account for up to one-third delays
```

Comment:
How does one-third related to "2x"?

Reply:
Yep, slightly low info in comments. First, 30 is a magic number in statistics. It's being said that if you measure something, your sample size should be >30 to be used. Or, in our case, if delay is more than 30% then it is a significant deviation from normal value. [Example](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://www.linkedin.com/pulse/magic-number-30-why-sample-size-often-considered-sufficient&ved=2ahUKEwiF5_Oo0YuIAxX4EkQIHUIFDWMQFnoECBQQAQ&usg=AOvVaw1lztfGKS4SRnuOHDnBTywc). Also, fun thing to watch is a [Simpson's paradox](https://www.youtube.com/watch?v=ebEkn-BiW5k) explanation.

Actual formula leading to this is this: 100 MB over 10MB/s line will be done in 10 sec. Same amount of data if we account for delays on processing, say throughput will drop to 7MB/s (30%), it will take ~14.5 sec. If we account for delays on the network and cut those 20% for it (Classic number for calculating network bandwidth to account for noisy neighbor's traffic) the resulting throughput will drop to 5MB/s (20% more), which is 20 sec.
That's why normal timeout is doubled.

Reply:
Could you please summarize this awesome explanation, and add to the comment?